macos/Sources/Helpers/Extensions/NSWindow+Extension.swift

```diff
@@ -15,4 +15,20 @@ extension NSWindow {
         guard let firstWindow = tabGroup?.windows.first else { return true }
         return firstWindow === self
     }
+
+    /// Adjusts the window origin if necessary to ensure the window remains visible on screen.
+    func constrainToScreen() {
+        guard let screen = screen ?? NSScreen.main else { return }
+        let visibleFrame = screen.visibleFrame
+        var windowFrame = frame
+
+        windowFrame.origin.x = max(visibleFrame.minX,
+            min(windowFrame.origin.x, visibleFrame.maxX - windowFrame.width))
```

Comment:
```suggestion
            min(windowFrame.origin.x, visibleFrame.width - windowFrame.width))
```

Could be some RTL issues here, but I don't know if Ghostty is fully compatible with that before. Need some testing to confirm.